src/Map.fs

```diff
@@ -11,50 +11,101 @@ type IMap =
-        Attributes.defineEnum<MapType>
-            "Map_MapType"
-            (fun _ newValueOpt node ->
-                let map = node.Target :?> Map
-
-                let value =
-                    match newValueOpt with
-                    | ValueNone -> MapType.Street
-                    | ValueSome v -> v
-
-                map.MapType <- value)
-            
-    let ItemsSource<'T> =
-        Attributes.defineBindable<WidgetItems<'T>, System.Collections.Generic.IEnumerable<Widget>>
-            Map.ItemsSourceProperty
-            (fun modelValue ->
-                seq {
-                    for x in modelValue.OriginalItems do
-                        modelValue.Template x
-                })
-            (fun a b -> ScalarAttributeComparers.equalityCompare a.OriginalItems b.OriginalItems)
 
+    let MoveToLastRegionOnLayoutChange =
+        Attributes.defineBindableBool Map.MoveToLastRegionOnLayoutChangeProperty
+
+    let RequestedRegion =
+        Attributes.defineSimpleScalarWithEquality<MapSpan> "Map_RequestedRegion" (fun _ newValueOpt node ->
+            let map = node.Target :?> Map
+
+            match newValueOpt with
+            | ValueNone -> ()
+            | ValueSome mapSpan -> map.MoveToRegion(mapSpan))
+
+    let Pins =
+        Attributes.defineListWidgetCollection "Map_Pins" (fun target -> (target :?> Map).Pins)
+
+    let MapClicked =
+        Attributes.defineEvent<MapClickedEventArgs> "Map_MapClicked" (fun target -> (target :?> Map).MapClicked)
 
 [<AutoOpen>]
 module MapBuilders =
     type Fabulous.XamarinForms.View with
 
         /// Defines a Map widget
-        static member inline Map<'msg>() =
-            WidgetBuilder<'msg, IMap>(Map.WidgetKey, AttributesBundle(StackList.empty (), ValueNone, ValueNone))
+        static member inline Map<'msg>(requestedRegion: MapSpan) =
+            CollectionBuilder<'msg, IMap, IPin>(Map.WidgetKey, Map.Pins, Map.RequestedRegion.WithValue(requestedRegion))
 
 [<Extension>]
 type MapModifiers =
+
+    [<Extension>]
+    static member inline hasZoomEnabled(this: WidgetBuilder<'msg, #IMap>, value: bool) =
+        this.AddScalar(Map.HasZoomEnabled.WithValue(value))
+
+    [<Extension>]
+    static member inline hasScrollEnabled(this: WidgetBuilder<'msg, #IMap>, value: bool) =
+        this.AddScalar(Map.HasScrollEnabled.WithValue(value))
+
+    [<Extension>]
+    static member inline mapType(this: WidgetBuilder<'msg, #IMap>, value: MapType) =
+        this.AddScalar(Map.MapType.WithValue(value))
+
+    [<Extension>]
+    static member inline isShowingUser(this: WidgetBuilder<'msg, #IMap>, value: bool) =
+        this.AddScalar(Map.IsShowingUser.WithValue(value))
+
+    [<Extension>]
+    static member inline isTrafficEnabled(this: WidgetBuilder<'msg, #IMap>, value: bool) =
+        this.AddScalar(Map.TrafficEnabled.WithValue(value))
+
+    [<Extension>]
+    static member inline moveToLastRegionOnLayout(this: WidgetBuilder<'msg, #IMap>, value: bool) =
```

Comment:
Small change to the name: `moveToLastRegionOnLayoutChange`

Reply:
Updated